server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -134,6 +146,51 @@ public CompletionStage<Result> review(
     return editOrReview(request, applicantId, programId, blockId, true);
   }
 
+  /** This method handles the applicant's selection from the address correction options. */
+  @Secure
+  public CompletionStage<Result> confirmAddress(
+      Request request, long applicantId, long programId, String blockId, boolean inReview) {
+
+    DynamicForm form = formFactory.form().bindFromRequest(request);
+    String selectedAddress = form.get(AddressCorrectionBlockView.SELECTED_ADDRESS_NAME);
+    Optional<String> maybeAddressJson = request.session().get(ADDRESS_JSON_SESSION_KEY);
+
+    ImmutableList<AddressSuggestion> suggestions =
+        addressSuggestionJsonSerializer.deserialize(
+            maybeAddressJson.orElseThrow(() -> new RuntimeException("Address JSON missing")));
+
+    return checkApplicantAuthorization(profileUtils, request, applicantId)
+        .thenComposeAsync(
+            v -> applicantService.getReadOnlyApplicantProgramService(applicantId, programId),
+            httpExecutionContext.current())
```

Comment:
**question (blocking):** is this stage needed? the `roApplicantProgramService` is not used in the next stage

Reply:
removed